### PR TITLE
DBCON-312: Close the JDBC Statement atOracleDbConnection.getTables()

### PR DESCRIPTION
**Pull request by niqueco**

### CODE REVIEW

File: src/main/java/org/mule/extension/db/internal/domain/connection/oracle/OracleDbConnection.java [hunk 1]
```diff
@@ -122,15 +122,16 @@ public OracleDbConnection(Connection jdbcConnection, List<DbType> customDataType
...
+      ResultSet resultSet = statement.getResultSet();
 
-    Set<String> tables = new HashSet<>();
-    while (resultSet.next()) {
-      tables.add(resultSet.getString(1));
+      Set<String> tables = new HashSet<>();
+      while (resultSet.next()) {
+        tables.add(resultSet.getString(1));
+      }
+      return tables;
```
Comment (juldiazmule) [inline src/main/java/org/mule/extension/db/internal/domain/connection/oracle/OracleDbConnection.java:133]: Add a line-break between "}" and "return"

**Comment by juldiazmule** [comment]
- please create a Jira ticket under DBCON project to track this change
- add a simple functional test that verifies if method "close" is called on the mocked statement

File: src/test/java/org/mule/extension/db/internal/domain/connection/OracleDbConnectionTestCase.java [hunk 2]
```diff
@@ -12,9 +12,7 @@
 import org.mule.db.commons.internal.domain.type.ResolvedDbType;
 import org.mule.tck.junit4.AbstractMuleTestCase;
 
-import java.sql.Connection;
-import java.sql.PreparedStatement;
-import java.sql.ResultSet;
+import java.sql.*;
```
Comment (nicomz) [inline src/test/java/org/mule/extension/db/internal/domain/connection/OracleDbConnectionTestCase.java:15]: No wildcard imports.

File: src/test/java/org/mule/extension/db/internal/domain/connection/OracleDbConnectionTestCase.java [hunk 3]
```diff
@@ -120,7 +118,23 @@ public void lobResolutionPerformance() throws Exception {
     assertThat(dbTypeCache.get(USER_TYPE_NAME_B).get(0).getName(), is(USER_TYPE_DBNAME_B));
     assertThat(dbTypeCache.keySet().size(), is(2));
     verify(preparedStatement, times(2)).executeQuery();
+  }
+
+  @Test
+  public void getTablesClosesTheStatement() throws SQLException {
```
Comment (nicomz) [inline src/test/java/org/mule/extension/db/internal/domain/connection/OracleDbConnectionTestCase.java:124]: Please run formatter.

**Review by nicomz — CHANGES_REQUESTED** [review]

**Review by juldiazmule — DISMISSED** [review]

**Review by nicomz — DISMISSED** [review]

**Review by nicomz — APPROVED** [review]

**Pull request merged**